src/components/swap/SwapBuyFiatButton.test.tsx

```diff
@@ -0,0 +1,127 @@
+import userEvent from '@testing-library/user-event'
+import { useWalletDrawer } from 'components/WalletDropdown'
+import { fireEvent, render, screen } from 'test-utils'
+
+import { useFiatOnrampAvailability, useOpenModal } from '../../state/application/hooks'
+import SwapBuyFiatButton, { MOONPAY_REGION_AVAILABILITY_ARTICLE } from './SwapBuyFiatButton'
+
+jest.mock('@web3-react/core', () => {
+  const web3React = jest.requireActual('@web3-react/core')
+  return {
+    ...web3React,
+    useWeb3React: () => ({
+      account: undefined,
+      isActive: false,
+    }),
+  }
+})
+
+jest.mock('../../state/application/hooks')
+const mockUseFiatOnrampAvailability = useFiatOnrampAvailability as jest.MockedFunction<typeof useFiatOnrampAvailability>
+const mockUseOpenModal = useOpenModal as jest.MockedFunction<typeof useOpenModal>
+
+jest.mock('components/WalletDropdown')
+const mockUseWalletDrawer = useWalletDrawer as jest.MockedFunction<typeof useWalletDrawer>
+
+const mockUseFiatOnRampsUnavailable = (shouldCheck: boolean) => {
+  if (shouldCheck) {
+    return {
+      available: false,
+      availabilityChecked: true,
+      error: null,
+      loading: false,
+    }
+  } else {
+    return {
+      available: false,
+      availabilityChecked: false,
+      error: null,
+      loading: false,
+    }
+  }
+}
+
+const mockUseFiatOnRampsAvailable = (shouldCheck: boolean) => {
+  if (shouldCheck) {
+    return {
+      available: true,
+      availabilityChecked: true,
+      error: null,
+      loading: false,
+    }
+  } else {
```

Comment:
you don't need the `else` statement here if you're returning in the first `if`

see [here](https://github.com/airbnb/javascript#blocks--no-else-return)

Reply:
thats true, removed